src/librustdoc/html/render/print_item.rs

```diff
@@ -643,6 +643,25 @@ fn item_function(cx: &Context<'_>, it: &clean::Item, f: &clean::Function) -> imp
     })
 }
 
+/// Struct used to handle insertion of "negative impl" marker in the generated DOM.
+struct NegativeMarker {
+    inserted_negative_marker: bool,
+}
```

Comment:
Currently we're generating the negative impl marker for every trait, even ones that never have negative impls.

An alternative (but slightly more complex) approach would be to only generate it if the first impl is negative (this would require adding a new state to the state machine), then generate it on-demand in `implementorsElems`.  This would avoid increasing the size of every documentation bundle by a factor of `number_of_traits`.

Reply:
I guess it's probably fine since the average crate only has a handful of traits, though I am a bit curious what impact this has on documentation bundle sizes of trait heavy crates like `num`.  My rough mental estimate is it would be something on the order of 100B, which seems acceptable.